libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/CodegenLayout.kt

```diff
@@ -147,14 +147,26 @@ internal abstract class CodegenLayout(
     return "${name.capitalizeFirstLetter()}Builder"
   }
 
-  fun mapName(name: String): String {
+  fun unknownBuilderName(name: String): String {
+    return "Other${name.capitalizeFirstLetter()}Builder"
+  }
+
+  fun objectMapName(name: String): String {
     return "${name.capitalizeFirstLetter()}Map"
   }
 
-  fun builderFunName(name: String): String {
+  fun unknownMapName(name: String): String {
```

Comment:
I don't think we need to distinguish between `OtherAnimalMap` and just `AnimalMap`. 

I would keep "other" for the public API (i.e. the `buildOtherAnimal`) because it's special (it requires typename). For everything else, I'd keep things symmetrical and just use `${graphqlType}Map` (so `AnimalMap` here for an example). Thoughts?

Reply:
Makes a lot of sense 👍 
Then my only objection there is to rename all `unkown` to `other`